blog-posts/takeuntil-takewhile/takeWhile-takeUntil-takeWhat.md

```diff
@@ -0,0 +1,106 @@
+---
+published: false
+title: "TakeWhile, TakeUntil, TakeWhat?"
+cover_image: ""
+description: "The RxJS TakeWhile and TakeUntil operator already sound similar and behave similar. So what's the difference?"
+tags: rxjs, javascript, typescript, webdev
+series:
+canonical_url:
+---
+
+It's already well known, that there are situations where you actually want to unsubscribe from your Observables.
+There are several
+ways to do this. In several projects, I stumbled across a mixture of using [`takeWhile`](https://rxjs.dev/api/operators/takeWhile) & [`takeUntil`](https://rxjs.dev/api/operators/takeUntil). This raises the question, why are there two of that, which sounds similar and act similarly?
+
+## About TakeWhile
+
+Well, let's take a look at a quick example. The first code snippet we are looking at is using `takeWhile` to unsubscribe from an observable. I used an Angular Component here, to couple the lifecycle of the observable to the lifecycle of the component.
+Short disclaimer: It doesn't necessarily need to be an Angular component, it's just the framework I'm the most familiar with.
```

Comment:
Using angular code will make this an angular article in the eyes of most readers. I understand this is the easiest to come up with an actual example,  but please consider making it rx code purely.

Reply:
Maybe add a small sentence in an extra box that tells non angular users what ngOnInit and ngondestroy do to understand the case here

Reply:
Or rather use a plain JavaScript example? Maybe using something like `mousedown` or `mousemove` events which you don't want anymore after release the mouse click (`mouseup`). Using "dragging an html element" or "drawing a line" as an example.